### DISCUSSION TITLE
how to convert PNG image to C array

### DISCUSSION THREAD

**Discussion by patricegamakuoh** [lovyan03/LovyanGFX#311] [Q&A] [answered]
How can I get PNG image converted into the C array to use with the lovyanGFX library as shown in this example? 
[https://github.com/lovyan03/LovyanGFX/tree/master/examples/Sprite/PartyParrot] 
https://github.com/lovyan03/LovyanGFX/tree/master/examples/Sprite/PartyParrot

is there a special converter or any suggestions regarding how to get my image into this format? as I'm not using an SD card, 
 thanks in advance!

**Comment by ciniml** [comment, marked as answer]
You can use `xxd` command if you are using Linux or WSL on Windows.

```shell
$ xxd -i foo.png foo.h
```

The file `foo.h` contains C style array which represents the contents of `foo.png`

**Reply by patricegamakuoh** [reply]
thanks for the quick feedback @ciniml  I have done the convertion with linux to get the resulting .h file then replaced the hex string with one of the parrot array in this code: https://github.com/lovyan03/LovyanGFX/tree/master/examples/Sprite/PartyParrot but nothing shows in the screen,,,

The size of my image is 319.7kb , the final .h array legnth is  319722, 
the display screen is 480x320  

Am i missing something?

**Reply by tobozo** [reply]
@patricegamakuoh it's hard to say without seeing the actual code and the original image (please create a gist)

some questions come to mind though:

- party parrot code uses bmp format but you mention png, did you adapt the functions?
- sprite size is capped by memory, have you tried with smaller images, different formats?
- screen rotation may affect rendering, have you tried different orientations?

**Comment by patricegamakuoh** [comment]
Thanks @tobozo you are right , i have converted to .bmp and it works well with @ciniml mentioned convertion method.

Another probblem now is the color performance , it does not seems to show the right image as expected .

how can i get the right image color output or what am i missing? 
thanks !

Original image
![original_image](https://user-images.githubusercontent.com/32825004/202119690-8a8c1e03-60eb-4ee5-bf55-c7d94b004cf5.png)

Final output image 
![output_image_from_display](https://user-images.githubusercontent.com/32825004/202119708-7a031a18-d992-47b2-9093-2f99ebe91c31.jpg)

**Comment by tobozo** [comment]
oh geez :) 

it's still hard to say without seeing the actual code, but looking at RGB/BGR situation I'd try to use `tft.setSwapBytes( bool )` and/or `sprite[x].setSwapBytes( bool )`

**Reply by patricegamakuoh** [reply]
thanks @tobozo  the prolblem was related to color inverter , and i just need to set the inverter paramater to true in the source code .
